kernel/timeout.c

```diff
@@ -18,6 +18,17 @@ static sys_dlist_t timeout_list = SYS_DLIST_STATIC_INIT(&timeout_list);
 
 static struct k_spinlock timeout_lock;
 
+/* On multiprocessor setups, it's possible to have multiple
+ * sys_clock_announce() calls arrive in parallel (the latest to exit
+ * the driver will generally be announcing zero ticks).  But we want
+ * the list of timeouts to be executed serially, so as not to confuse
+ * application code.  This lock wraps the announce loop, external to
+ * the nested timeout_lock.
+ */
+#ifdef CONFIG_SMP && CONFIG_MP_NUM_CPUS > 1
```

Comment:
I've seen this idiom in multiple places now, is it time to create a new Kconfig option that encapsulates it?

Reply:
Probably.  It's really just a case of overoptimization.  In the real world it's going to be extremely rare to have SMP=y but only one cpu.  But we do that in test code a lot for coverage, so I'm hypersensitive to the distinction.  Really this could just read "#ifdef CONFIG_SMP".

Oh... but heh: it's a bug anyway!  I need '#if defined(CONFIG_SMP)'